lib/counter.js

```diff
@@ -25,34 +24,24 @@ const {
 class Counter {
 	/**
 	 * Counter
-	 * @param {string} name - Name of the metric
-	 * @param {string} help - Help description for the metric
-	 * @param {Array.<string>} labels - Labels
+	 * @param {config} config - Configuration object.
+	 * @param {string} config.name - Name of the metric
+	 * @param {string} config.help - Help description for the metric
+	 * @param {Array.<string>} config.labels - Array with strings, all label keywords supported
 	 */
-	constructor(name, help, labels) {
-		let config;
-		if (isObject(name)) {
-			config = Object.assign(
-				{
-					labelNames: []
-				},
-				name
-			);
-
-			if (!config.registers) {
-				config.registers = [globalRegistry];
-			}
-		} else {
-			printDeprecationObjectConstructor();
-
-			config = {
-				name,
-				help,
-				labelNames: labels,
-				registers: [globalRegistry]
-			};
+	constructor(config) {
+		if (!isObject(config)) {
+			throw new Error('constructor expected a config object');
+		}
+		config = Object.assign(
```

Comment:
We can use object spread

Reply:
Object.assign is used all over the place ATM, could someone else do js dialect updates later?

Reply:
of course 👍